lib/puppet/pal/catalog_compiler.rb

```diff
@@ -97,6 +97,15 @@ def evaluate_additions
       internal_compiler.evaluate_additions
     end
 
+    # Attempts to evaluate AST for node defnintions https://puppet.com/docs/puppet/latest/lang_node_definitions.html
+    # if there are any.
+    def evaluate_ast_node
+      # This is definitely calling an internal function in the compiler class, we could update
+      # Puppet::Parser::Compiler to make that a public function, but I was concerned about
+      # changing something that could potentially be seen as public puppet API.
+      #                - Sean P. McDonald 5/7/2020
+      internal_compiler.send(:evaluate_ast_node)
```

Comment:
This is something we can change to be an actual public API if we want, I was just concerned about making more things public in puppet.

Reply:
We can make it Ruby `public` while documenting it as `# @api private`. There's some prior art to doing that in the rest of Puppet.